webgpu.h

```diff
@@ -752,12 +786,14 @@ typedef struct WGPUBindGroupDescriptor {
     WGPUBindGroupEntry const * entries;
 } WGPUBindGroupDescriptor;
 
-typedef struct WGPUBindGroupLayoutDescriptor {
-    WGPUChainedStruct const * nextInChain;
-    char const * label;
-    uint32_t entryCount;
-    WGPUBindGroupLayoutEntry const * entries;
-} WGPUBindGroupLayoutDescriptor;
+typedef struct WGPUBindGroupLayoutEntry {
+    uint32_t binding;
+    WGPUShaderStageFlags visibility;
+    WGPUBufferBindingLayout buffer;
```

Comment:
Could this be a `union`?

Reply:
It could be it would need a tag, and unions in general are pretty unsafe in C and C++, plus this is pretty cold code. We had some discussion about this topic with @kainino0x and @toji and came to the solution in this PR.

Reply:
Another thing I don't understand is each of `WGPUBufferBindingLayout`, `WGPUSamplerBindingLayout`, and `WGPUTextureBindingLayout` being extendable, but `WGPUBindGroupLayoutEntry` itself - not extendable. Would it be easier to make it the opposite?

Reply:
Uh, WGPUBindGroupLayoutEntry should definitely be extendable, that's a mistake. It might be useful to have eachsub-structure are extendable but I see how that could be unnecessary and using space for nothing. We could probably remove it. @toji WDYT?

Reply:
I think both should be extendable, it's a small overhead (cold code, like you said) and I think it's likely that most extensions would show up in the inner structures.

Reply:
I don't have anything against it being extendable.

Reply:
Made it extendable.